hack/preview.sh

```diff
@@ -71,10 +71,13 @@ fi
 git checkout -b $PREVIEW_BRANCH
 
 # patch argoCD applications to point to your fork
-oc kustomize $ROOT/argo-cd-apps/overlays/development | yq e "select(.spec.source.repoURL==\"https://github.com/redhat-appstudio/infra-deployments.git\")
-  | del(.spec.destination, .spec.syncPolicy, .spec.project, .spec.source.path, .metadata.namespace)
-  | .spec.source.repoURL=\"$MY_GIT_REPO_URL\"
-  | .spec.source.targetRevision=\"$PREVIEW_BRANCH\"
+oc kustomize $ROOT/argo-cd-apps/overlays/development | yq e "select(.spec.source.repoURL==\"https://github.com/redhat-appstudio/infra-deployments.git\" or .spec.template.spec.source.repoURL==\"https://github.com/redhat-appstudio/infra-deployments.git\")
+  | del(.metadata.namespace, .spec.destination, .spec.syncPolicy, .spec.project, .spec.source.path, .spec.template.spec.destination, .spec.template.spec.syncPolicy, .spec.template.spec.project, .spec.template.spec.source.path)
+  | (.spec | select(has(\"source\")) | .source.repoURL)=\"$MY_GIT_REPO_URL\"
+  | (.spec | select(has(\"source\")) | .source.targetRevision)=\"$PREVIEW_BRANCH\"
+  | (.spec | select(has(\"template\")) | .template.spec.source.repoURL)=\"$MY_GIT_REPO_URL\"
```

Comment:
Is this change needed to take into account the ApplicationSet structure?

Reply:
Yeah, I tried to do it using `with` but lucked out